### PR TITLE
Add documentation for build pruning

### PR DESCRIPTION
**Pull request by coreydaley**
Provides supporting documentation for https://github.com/openshift/origin/pull/13788 and https://github.com/openshift/origin/pull/14375

@openshift/devex ptal

### CODE REVIEW

File: dev_guide/builds/advanced_build_operations.adoc [hunk 1]
```diff
@@ -262,3 +262,31 @@ whenever the first build is run and produces a new image with the binary
 artifact, the second build is automatically triggered to produce a runtime image
 that contains that artifact. Therefore, both builds behave as a single build
 with two stages.
+
+[[build-pruning]]
+== Build Pruning
+
+By default, builds that have completed their lifecycle are persisted indefinitely.
+You can limit the number of previous builds that are kept around by supplying a positive
```
Comment (bparees) [inline dev_guide/builds/advanced_build_operations.adoc]: s/kept around/retained/

File: dev_guide/builds/advanced_build_operations.adoc [hunk 2]
```diff
@@ -262,3 +262,31 @@ whenever the first build is run and produces a new image with the binary
...
+apiVersion: "v1"
+kind: "BuildConfig"
+metadata:
+  name: "sample-build"
+spec:
+  successfulBuildsHistoryLimit: 2 <1>
+  failedBuildsHistoryLimit: 2 <2>
+----
+<1> `successfulBuildsHistoryLimit` will prune builds with a status of `completed`
+<2> `failedBuildsHistoryLimit` will prune builds with a status of `failed`, `cancelled`, or `error`
```
Comment (bparees) [inline dev_guide/builds/advanced_build_operations.adoc]: "will retain up to 2 builds with a status of completed"
"will retain up to 2 builds with a status of failed, cancelled, or error"

File: dev_guide/builds/advanced_build_operations.adoc [hunk 3]
```diff
@@ -262,3 +262,31 @@ whenever the first build is run and produces a new image with the binary
...
+
+Build pruning is triggered by the following actions:
+
+ - the build configuration is updated
+ - a build completes it's lifecycle
```
Comment (bparees) [inline dev_guide/builds/advanced_build_operations.adoc]: s/it's/its/

**Review by bparees — CHANGES_REQUESTED** [review]

**Comment by coreydaley** [comment]
@bparees updated

File: dev_guide/builds/advanced_build_operations.adoc [hunk 4]
```diff
@@ -262,3 +262,31 @@ whenever the first build is run and produces a new image with the binary
...
+----
+apiVersion: "v1"
+kind: "BuildConfig"
+metadata:
+  name: "sample-build"
+spec:
+  successfulBuildsHistoryLimit: 2 <1>
+  failedBuildsHistoryLimit: 2 <2>
+----
+<1> `successfulBuildsHistoryLimit` will retain up to 2  builds with a status of `completed`
```
Comment (bparees) [inline dev_guide/builds/advanced_build_operations.adoc]: somehow you put an extra space after the "2" :)

**Review by bparees — CHANGES_REQUESTED** [review]

**Comment by coreydaley** [comment]
@bparees fixed

**Comment by bparees** [comment]
please cross-link this new section and the existing build pruning section to each other:
https://docs.openshift.org/latest/admin_guide/pruning_resources.html#pruning-builds

people reading about one technique will probably want to be aware of the existence of the other technique.

**Comment by coreydaley** [comment]
@bparees updated

**Comment by bparees** [comment]
lgtm. @openshift/team-documentation ptal/merge.

**Comment by bmcelvee** [comment]
LGTM

**Comment by bmcelvee** [comment]
[rev_history]
|xref:..admin_guide/pruning_resources.adoc#admin-guide-pruning-resources[Pruning Objects]
|Added note to xref:..admin_guide/pruning_resources.adoc#pruning-builds[Pruning Builds] linking to the xref:../dev_guide/builds/advanced_build_operations.adoc#build-pruning[Build Pruning] section.
%
|xref:../dev_guide/builds/advanced_build_operations.adoc#dev-guide-advanced-build-operations[Advanced Build Operations]
|Added the xref:../dev_guide/builds/advanced_build_operations.adoc#build-pruning[Build Pruning] section.
%

**Comment by bparees** [comment]
@bmcelvee this doc appears to have found its way into 3.4/3.5:
https://docs.openshift.com/container-platform/3.4/dev_guide/builds/advanced_build_operations.html#build-pruning

it should not have (the feature is 3.6+)

**Comment by vikram-redhat** [comment]
@bparees sorry about that. It has now been removed:

3.4: https://docs.openshift.com/container-platform/3.4/dev_guide/builds/advanced_build_operations.html

3.5: https://docs.openshift.com/container-platform/3.5/dev_guide/builds/advanced_build_operations.html

**Pull request merged**